emulator/mgba/src/vfile/shared.rs

```diff
@@ -0,0 +1,64 @@
+use std::{
+    io::{Read, Seek, Write},
+    sync::{Arc, Mutex},
+};
+
+use super::VFile;
+
+pub struct Shared<V> {
```

Comment:
It's probably a terrible idea to modify this if mgba has it, randomly seeking around can't be a good idea! It could lead to unsoundness, I imagine. But it is useful to be able to recover the buffer afterwards... That's the abstraction I want to create.